### PR TITLE
#71 add restore password logic

### PR DESCRIPTION
**Pull request by Eugene204**

### CODE REVIEW

File: src/app/components/after-restore-password/after-restore-password.component.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  ngOnInit(): void {
+   this.restorePassword();
+  }
+  
+  restorePassword=()=>
+  {
+    const token = this.route.snapshot.queryParams['token'];
+    const email = this.route.snapshot.queryParams['email'];
+    const password = this.route.snapshot.queryParams['password'];
+
```
Comment (LizaShemetovska) [inline src/app/components/after-restore-password/after-restore-password.component.ts]: delete empty line

File: src/app/components/restore-password/restore-password.component.css [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    margin-bottom:10px;
+}
+.error_alert_restorePassword {
+    width: 30%;
+}
+
+#center-restorePassword {
+    display: flex;
+    justify-content: center;
+}
```
Comment (LizaShemetovska) [inline src/app/components/restore-password/restore-password.component.css]: add empty line

File: src/app/models/restorePassword.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (LizaShemetovska) [inline src/app/models/restorePassword.ts]: delete empty line

File: src/app/services/account.service.ts [hunk 4]
```diff
@@ -48,7 +50,19 @@ export class AccountService {
...
+  forgotPassword(data: RestorePassword ) {
+    console.log(data);
+    data.passwordURI = RESTORE_PASSWORD_URI;
+    return this.http.post<any>(`${HOST_URL}Account/ForgotPassword`, data, httpOptions);
+  }
+
+  restorePassword(route:string,token:string,email:string,password:string) {
+    return this.http.get<any>(`${HOST_URL}${route}/${password}/${token}/${email}`);
+  } 
+
```
Comment (LizaShemetovska) [inline src/app/services/account.service.ts]: delete empty line

File: src/app/components/restore-password/restore-password.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    .subscribe((data: any) => {
+      this.router.navigate(['/home']);
+      this.toastr.success('Password successfully changed', 'Congratulations');
+    },
+    err => {
+      this.errorMessage = err.error.message;
+      this.isRestorePasswordFailed = true;
+    });
+  }
+
```
Comment (LizaShemetovska) [inline src/app/components/restore-password/restore-password.component.ts]: delete empty line

File: package.json [hunk 6]
```diff
@@ -10,15 +10,17 @@
...
-    "@angular/animations": "~12.2.0",
+    "@angular/animations": "^12.2.5",
     "@angular/common": "~12.2.0",
     "@angular/compiler": "~12.2.0",
     "@angular/core": "~12.2.0",
     "@angular/forms": "~12.2.0",
     "@angular/platform-browser": "~12.2.0",
     "@angular/platform-browser-dynamic": "~12.2.0",
     "@angular/router": "~12.2.0",
+    "angular-toastr": "^2.1.1",
```
Comment (LizaShemetovska) [inline package.json:21]: delete this package

**Review by LizaShemetovska — CHANGES_REQUESTED** [review]

**Review by LizaShemetovska — APPROVED** [review]

**Review by ICHTLAY — APPROVED** [review]

**Pull request merged**